scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -417,12 +424,12 @@ object Defn {
   @ast class Def(
       mods: List[Mod],
       name: Term.Name,
-      tparams: List[scala.meta.Type.Param],
-      paramss: List[List[Term.Param]],
+      paramss: List[Clause],
       decltpe: Option[scala.meta.Type],
       body: Term
   ) extends Defn with Member.Term with Stat.WithMods {
-    checkFields(paramss.forall(onlyLastParamCanBeRepeated))
+    val termParamss = paramss.collect{ case Clause.TermClause(termParams) => termParams }
+    checkFields(paramss.collect{ case Clause.TermClause(termParams) => termParams }.forall(onlyLastParamCanBeRepeated))
```

Comment:
```suggestion
    checkFields(paramss.forall {
      case term: Clause.TermClause =>
        onlyLastParamCanBeRepeated(term.params)
      case _ => true
    })
```
Macros seem to be breaking if unapply is used inside here

Reply:
Thank you so much, it was that !